cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go

```diff
@@ -43,7 +52,41 @@ func (r *ImageMapReader) Read(ctx context.Context) ([]common.ImageMapItem, error
 		return []common.ImageMapItem{}, err
 	}
 
-	return []common.ImageMapItem{}, nil // TODO: Implement.
+	err = r.readBAT(ctx)
+	if err != nil {
+		return []common.ImageMapItem{}, err
+	}
+
+	var items []common.ImageMapItem
+	var item common.ImageMapItem
+	var entry imageMapEntry
+
+	for i, _ := range r.bat {
+		offset := item.Start + item.Length
+		bytes := min(uint64(r.header.BlockSize), r.Size()-offset)
```

Comment:
What does the `bytes` mean? Please, come up with another naming. Consider entrySizeBytes or something. It is a size.